website_attribute_set/controllers/main.py

```diff
@@ -0,0 +1,421 @@
+# Copyright 2011 Kencove (http://www.kencove.com).
+# @author Mohamed Alkobrosli <malkobrosly@kencove.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+
+from werkzeug.exceptions import NotFound
+
+from odoo import fields
+from odoo.http import request, route
+from odoo.models import BaseModel
+from odoo.osv import expression
+from odoo.tools import SQL, float_round, groupby, lazy
+
+from odoo.addons.website.controllers.main import QueryURL
+from odoo.addons.website_sale.controllers import main
+
+
+class WebsiteSale(main.WebsiteSale):
+    @route(
+        [
+            "/shop",
+            "/shop/page/<int:page>",
+            '/shop/category/<model("product.public.category"):category>',
+            '/shop/category/<model("product.public.category"):category>/page/<int:page>',
+        ],
+        type="http",
+        auth="public",
+        website=True,
+        sitemap=main.WebsiteSale.sitemap_shop,
+    )
+    def shop(
+        self,
+        page=0,
+        category=None,
+        search="",
+        min_price=0.0,
+        max_price=0.0,
+        ppg=False,
+        **post,
+    ):
+        if not request.website.has_ecommerce_access():
+            return request.redirect("/web/login")
```

Comment:
```suggestion
            return request.redirect("/web/login")
```
should call super() in here?

Reply:
there is no super in this method, I am overriding the entire method by the way!

Reply:
I think we should not destroy the original features too much but only intervene in a certain processing.